workspaces/liferay-ticket-workspace/client-extensions/liferay-ticket-custom-element/src/pages/TicketsByStatusDashboard.tsx

```diff
@@ -0,0 +1,214 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import {DndContext, PointerSensor, useSensor, useSensors} from '@dnd-kit/core';
+import React, {useEffect, useState} from 'react';
+import {QueryClient, useMutation} from 'react-query';
+
+import StatusColumn from '../components/StatusColumn';
+import {useTickets} from '../hooks/useTickets';
+import {Liferay} from '../services/liferay';
+import {
+	J3Y7_STATUSES,
+	fetchListTypeDefinitions,
+} from '../services/listTypeEntries';
+import {updateTicketStatus} from '../services/tickets';
+
+import '../styles/TicketsByStatusDashboard.css';
+
+const DRAG_RESULT = {
+	NO_CHANGE: 'NO_CHANGE',
+	STATUS_CHANGED: 'STATUS_CHANGED',
+};
+
+const allowedStatusesForDashboard = [
+	{
+		key: 'open',
+		order: 1,
+	},
+	{
+		key: 'inProgress',
+		order: 2,
+	},
+	{
+		key: 'answered',
+		order: 3,
+	},
+	{
+		key: 'closed',
+		order: 4,
+	},
+];
+
+function statusSortFunction(a: any, b: any) {
+	const aEntry = allowedStatusesForDashboard.find(
+		(temp) => temp.key === a.key
+	);
+	const bEntry = allowedStatusesForDashboard.find(
+		(temp) => temp.key === b.key
+	);
+
+	if (aEntry && bEntry) {
+		return aEntry.order - bEntry.order;
+	}
+
+	return 0;
+}
```

Comment:
Allowed statuses and a status sort function seem more complicated than necessary. Let's discuss the problem on a call and talk through a solution.